lib/services.template.ejs

```diff
@@ -469,77 +521,103 @@ if (action.getHttpMethod() == 'POST' || action.getHttpMethod() == 'PUT') {
 <%   });
   }
 } -%>
-         *
+           *
 <% var returnType = action.isReturningArray() ? 'Array.<Object>': 'Object'; -%>
-         * @param {function(<%- returnType %>,Object)=} successCb
-         *   Success callback with two arguments: `value`, `responseHeaders`.
-         *
-         * @param {function(Object)=} errorCb Error callback with one argument:
-         *   `httpResponse`.
-         *
-         * @returns {<%- returnType %>} An empty reference that will be
-         *   populated with the actual data once the response is returned
-         *   from the server.
-         *
+           * @param {function(<%- returnType %>,Object)=} successCb
+           *   Success callback with two arguments: `value`, `responseHeaders`.
+           *
+           * @param {function(Object)=} errorCb Error callback with one argument:
+           *   `httpResponse`.
+           *
+           * @returns {<%- returnType %>} An empty reference that will be
+           *   populated with the actual data once the response is returned
+           *   from the server.
+           *
 <% if (!action.returns || action.returns.length == 0) { -%>
-         * This method returns no data.
+           * This method returns no data.
 <% } else if (action.returns[0].root) { -%>
 <%   if (action.returns[0].description) { -%>
-         * <%- action.returns[0].description
+           * <%- action.returns[0].description
 .replace(/\n/g, '\n         * ').trimRight() %>
 <%   } else { -%>
-         * <em>
-         * (The remote method definition does not provide any description.
-         * This usually means the response is a `<%- responseModelName || modelName %>` object.)
-         * </em>
+           * <em>
+           * (The remote method definition does not provide any description.
+           * This usually means the response is a `<%- responseModelName || modelName %>` object.)
+           * </em>
 <%   } -%>
 <% } else { -%>
-         * Data properties:
+           * Data properties:
 <%   action.returns.forEach(function(arg) { -%>
-         *
-         *  - `<%- arg.arg %>` – `{<%- getJsDocType(arg)  %>}` - <%-
+           *
+           *  - `<%- arg.arg %>` – `{<%- getJsDocType(arg)  %>}` - <%-
 (arg.description || '').replace(/\n/g, '\n         *   ') %>
 <%   });
    }
 -%>
-         */
+           */
 <% } // end of ngdocForMethod -%>
 
+/**
+ * @ngdoc overview
+ * @name <%- moduleName %>
+ * @module
+ * @description
+ *
+ * The `<%- moduleName %>` module provides services for interacting with
+ * the models exposed by the LoopBack server via the REST API.
+ *
+ */
+var module = angular.module(<%-: moduleName | q %>, [<%-: registryName | q %>]);
+
+<% for (var modelKey in models) { -%>
+(function(modelName) {
+  module.factory(modelName,
+  [
+    <%-: registryName | q %>,
+    'LoopBackAuth',
+    function(StrongLoopRegistry, LoopBackAuth) {
+      LoopBackAuth.switchContext(_urlBase);
+      return new StrongLoopRegistry(_urlBase)[modelName];
```

Comment:
Why is it necessary to switch contexts here? I would prefer `StrongLoopRegistry` to take care of prefixing any localStorage keys itself, for example

```js
return new StrongLoopRegistry(_urlBase, '$LoopBack$');
```

If possible, we should preserve backwards compatibility - new clients generated with this new codegen version should be still able to load auth credentials stored in localStorage by old clients.